cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/DeduplicateAggregationDefinition.java

```diff
@@ -0,0 +1,129 @@
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.aggregation;
+
+import io.cdap.cdap.etl.api.relational.Expression;
+
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Specifies how a deduplicate operation should be executed.
+ * A deduplication is an operation that takes in a set of rows and for each group of rows for which the values for
+ * certain group by fields are the same, outputs a single row.
+ * This single row can be determined by performing an operation on the set of rows which returns a single row.
+ * Currently the two supported operations are returning the row with the MIN or MAX value of a particular expression.
+ */
+public class DeduplicateAggregationDefinition extends AggregationDefinition {
+  List<SimpleEntry<Expression, FilterFunction>> filterExpressions;
+
+  private DeduplicateAggregationDefinition(List<Expression> groupByExpressions,
+                                           List<Expression> selectExpressions,
+                                           List<SimpleEntry<Expression, FilterFunction>> filterExpressions) {
+    super(groupByExpressions, selectExpressions);
+    this.filterExpressions = filterExpressions;
+  }
+
+  /**
+   * Get the list of expressions using which a single record will be selected from a group of records for deduplication.
+   * Each entry in the list is defined as a pair of an {@link Expression} and {@link FilterFunction}, specifying
+   * the function to be executed on the expression.
+   * @return A {@link List} of {@link SimpleEntry} objects, each of which is a mapping between an {@link Expression}
+   * and a {@link FilterFunction}.
+   */
+  public List<SimpleEntry<Expression, FilterFunction>> getFilterExpressions() {
+    return filterExpressions;
+  }
+
+  /**
+   * An enumeration that specifies the function to be executed on the filter expression for each group of rows
+   * to obtain a single row after deduplication.
+   */
+  public enum FilterFunction {
+    MIN, MAX
+  }
+
+  /**
+   * @return A builder to create a DeduplicateAggregationDefinition.
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Builds a DeduplicateAggregationDefinition using fields to dedup on, fields to select,
+   * and fields to use to filter the first record to be output when removing duplicates.
+   * The fields to dedup on and the fields to select must be specified.
+   */
+  public static class Builder {
+    private List<Expression> groupByExpressions;
+    private List<Expression> selectExpressions;
+    List<SimpleEntry<Expression, FilterFunction>> filterExpressions;
+
+    public Builder() {
+      groupByExpressions = Collections.emptyList();
+      selectExpressions = Collections.emptyList();
+      filterExpressions = new ArrayList<>();
+    }
+
+    public Builder dedupOn(List<Expression> dedupExpressions) {
```

Comment:
Please also add javadocs to the builder. It helps when working from IDE and building the definition

Reply:
Added.